### PR TITLE
Update Dockerfile to include Faiss library

### PR DESCRIPTION
**Pull request by PPPoint-t**
Modify Dockerfile to support faiss

### CODE REVIEW

File: ci/images/tugraph-compile-centos7-Dockerfile [hunk 1]
```diff
@@ -340,6 +340,18 @@ RUN wget https://tugraph-web.oss-cn-beijing.aliyuncs.com/tugraph/deps/tabulate-3
...
+# 安装 SWIG
+RUN wget https://tugraph-web.oss-cn-beijing.aliyuncs.com/tugraph/deps/swig-4.0.2.tar.gz \
+    && tar -xzvf swig-4.0.2.tar.gz && cd swig-4.0.2 && ./configure \
+    && make && make install && cd .. \
+    && rm -rf swig-4.0.2 swig-4.0.2.tar.gz
+
+# 安装 FAISS
+RUN wget -O /tmp/openblas-faiss.tar.gz https://tugraph-web.oss-cn-beijing.aliyuncs.com/tugraph/deps/openblas-faiss.tar.gz \
+    && tar -xzvf /tmp/openblas-faiss.tar.gz -C /tmp && cd /tmp/openblas-faiss/OpenBLAS && make && make install \
+    && cd /tmp/openblas-faiss/faiss && cmake -B build -DCMAKE_BUILD_TYPE=Release -DFAISS_ENABLE_GPU=OFF -DFAISS_OPT_LEVEL=avx2 -DBUILD_TESTING=OFF -DBUILD_SHARED_LIBS=OFF -DCMAKE_INSTALL_PREFIX=/usr/local -DCMAKE_INSTALL_LIBDIR=lib -DOpenMP_CXX_FLAGS="-fopenmp" -DOpenMP_CXX_LIB_NAMES="gomp" -DOpenMP_gomp_LIBRARY="/usr/lib64/libgomp.so.1" -DBLAS_LIBRARIES=/opt/OpenBLAS/lib/libopenblas.so . && make -C build && make -C build install \
```
Comment (ljcui) [inline ci/images/tugraph-compile-centos7-Dockerfile:352]: 最终的lgraph_server 会链接 libopenblas.so  这个动态库吗

**Comment by ljcui** [comment]
在centos7的编译环境下，ldd 下 lgraph_server ，贴下依赖了哪些动态库。

File: ci/images/tugraph-compile-centos7-Dockerfile (hunk 1 above)
Comment (PPPoint-t) [reply ci/images/tugraph-compile-centos7-Dockerfile:352]: 不会的，只是faiss的编译需要用openblas库

**Comment by PPPoint-t** [comment]
> 在centos7的编译环境下，ldd 下 lgraph_server ，贴下依赖了哪些动态库。
centos7环境下动态库依赖：
        linux-vdso.so.1 =>  (0x00007fff30b63000)
        libvsag.so => /usr/local/lib/libvsag.so (0x00007fa75228b000)
        liblgraph.so => /usr/local/lib/liblgraph.so (0x00007fa74fea1000)
        libpthread.so.0 => /usr/local/lib/libpthread.so.0 (0x00007fa74fc85000)
        librt.so.1 => /usr/local/lib/librt.so.1 (0x00007fa74fa7d000)
        libz.so.1 => /usr/local/lib/libz.so.1 (0x00007fa74f867000)
        libdl.so.2 => /usr/local/lib/libdl.so.2 (0x00007fa74f663000)
        libm.so.6 => /usr/local/lib/libm.so.6 (0x00007fa74f361000)
        libc.so.6 => /usr/local/lib/libc.so.6 (0x00007fa74ef93000)
        /lib64/ld-linux-x86-64.so.2 (0x00007fa752d83000)
        libgomp.so.1 => /usr/local/lib/libgomp.so.1 (0x00007fa74ed67000)
        libgfortran.so.5 => /usr/local/lib/libgfortran.so.5 (0x00007fa74e8ef000)
        libgcc_s.so.1 => /usr/local/lib/libgcc_s.so.1 (0x00007fa74e6d7000)
        libquadmath.so.0 => /usr/local/lib/libquadmath.so.0 (0x00007fa74e497000)

File: ci/images/tugraph-compile-centos7-Dockerfile [hunk 2]
```diff
@@ -340,6 +340,18 @@ RUN wget https://tugraph-web.oss-cn-beijing.aliyuncs.com/tugraph/deps/tabulate-3
     && tar zxf tabulate-3a5830.tar.gz && cp -rf tabulate/include/tabulate /usr/local/include \
     && rm -rf /tabulate*
 
+# 安装 SWIG
+RUN wget https://tugraph-web.oss-cn-beijing.aliyuncs.com/tugraph/deps/swig-4.0.2.tar.gz \
+    && tar -xzvf swig-4.0.2.tar.gz && cd swig-4.0.2 && ./configure \
+    && make && make install && cd .. \
```
Comment (ljcui) [inline ci/images/tugraph-compile-centos7-Dockerfile:346]: make 改成  make ${JFLAG}

File: ci/images/tugraph-compile-centos7-Dockerfile [hunk 3]
```diff
@@ -340,6 +340,18 @@ RUN wget https://tugraph-web.oss-cn-beijing.aliyuncs.com/tugraph/deps/tabulate-3
...
+    && rm -rf swig-4.0.2 swig-4.0.2.tar.gz
+
+# 安装 FAISS
+RUN wget -O /tmp/openblas-faiss.tar.gz https://tugraph-web.oss-cn-beijing.aliyuncs.com/tugraph/deps/openblas-faiss.tar.gz \
+    && tar -xzvf /tmp/openblas-faiss.tar.gz -C /tmp && cd /tmp/openblas-faiss/OpenBLAS && make && make install \
```
Comment (ljcui) [inline ci/images/tugraph-compile-centos7-Dockerfile:351]: -DFAISS_OPT_LEVEL=avx2，有些旧的机型没有avx2指令集, 换成 -DFAISS_OPT_LEVEL=generic

**Review by ljcui — APPROVED** [review]

**Pull request merged**